### ISSUE TITLE
I have no access to old orders (from 2019) PS 1.7.7.3

### ISSUE THREAD

**Issue by didi-2018** [PrestaShop/PrestaShop#24158]
#### Describe the bug
I try to open an old order from 2019 but I only get an error message `(see the first attached screenshot below)`
After enable Debug Mode I get the Errors in the video `(see the attached screen record below)`

https://drive.google.com/file/d/1ezyd3Q5Mr1GvwF5HkeqIVncMPs690CWM/view 

and disable Debug mode is not more possible `(see the second attached screenshot below)`
It is hard to follow your Template here for how to describe a bug cause I dont know to explain the steps. I will try to:

#### Steps to Reproduce

1. go to orders overview and search an order by typing customers name
2. try to open that order
3. see error message` (see the third attached screenshot below)`
4. enable Debus Mode
5. see Errors: https://drive.google.com/file/d/1ezyd3Q5Mr1GvwF5HkeqIVncMPs690CWM/view 
6. disable Debug Mode is not possible `(see the fourth attached screenshot below)`

Same happens when you go over the Customer account section in Backoffice
Does anybody have an idea how I now can access this old order? I need access cause it needs to be given to an lawyer. 

**Screenshots**
**1. Screenshot**

![image](https://user-images.githubusercontent.com/44971218/115711600-01671a00-a374-11eb-848d-a7614c84e914.png)

**2. Screenshot**

![image](https://user-images.githubusercontent.com/44971218/115711743-33787c00-a374-11eb-87c3-ec5b468e2988.png)

**3. Screenshot**

![image](https://user-images.githubusercontent.com/44971218/115711600-01671a00-a374-11eb-848d-a7614c84e914.png)

**4. Screenshot**

![image](https://user-images.githubusercontent.com/44971218/115711743-33787c00-a374-11eb-87c3-ec5b468e2988.png)

#### Additional information

- PS: 1.7.7.3 (update from 1.7.6.9) 
- PHP: 7.2.34

**Comment by didi-2018** [comment]
It happens for all Orders OLDER than 01.03.20 from 29.02.20 and older. 

Orders from 01.03.20 and up have not those problems. 

Can you please check this?

**Comment by matks** [comment]
Probably duplicate of https://github.com/PrestaShop/PrestaShop/issues/22862

**Comment by didi-2018** [comment]
@matks they report any orders. I report orders older than 29.02.20. So I do not think it is the same. I will check tomorrow if the time range changes from day to day. 

I've been thinking for a long time to update to 1.7.7.x or not. Than I decided to but I regrett it from the Beginning. Only problems. 

Can I go back to 1.7.6.9 and use the database from 1.7.7.3? That would be the best. I have not expected so many errors with 1.7.7.x, I thought it would be better.

**Comment by kmorgen** [comment]
Try to do this: 
https://tecserve.eu/prestashop17/prestashop-error-after-upgrading-1-7-7-1/

Open your database with your favorite tool (ex. phpMyAdmin) and paste all the mentiond code into your SQL command line and run it.

**Comment by prston** [comment]
@didi-2018 there is a full thread on issue #22862 in addition to @kmorgen link

https://github.com/PrestaShop/PrestaShop/issues/22862

**Comment by didi-2018** [comment]
@kmorgen thank you for the hint! 

This would be happen if I do this:

```
SQL-Befehl:
UPDATE uhqb_address SET phone = '' WHERE phone IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_address SET phone_mobile = '' WHERE phone_mobile IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_detail SET id_order_invoice = '0' WHERE id_order_invoice IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_detail SET product_reference = '' WHERE product_reference IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_detail SET product_supplier_reference = '' WHERE product_supplier_reference IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_detail SET product_isbn = '' WHERE product_isbn IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_detail SET product_upc = '' WHERE product_upc IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_detail SET product_mpn = '' WHERE product_mpn IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_detail SET download_hash = '' WHERE download_hash IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_orders SET gift_message = '' WHERE gift_message IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_payment SET transaction_id = '' WHERE transaction_id IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_payment SET card_number = '' WHERE card_number IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_payment SET card_brand = '' WHERE card_brand IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_payment SET card_expiration = '' WHERE card_expiration IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_payment SET card_holder = '' WHERE card_holder IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_carrier SET tracking_number = '' WHERE tracking_number IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_carrier SET id_order_invoice = '0' WHERE id_order_invoice IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_order_carrier SET weight = '0' WHERE weight IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product SET isbn = '' WHERE isbn IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product SET upc = '' WHERE upc IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product SET mpn = '' WHERE mpn IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product SET unity = '' WHERE unity IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product SET supplier_reference = '' WHERE supplier_reference IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product SET location = '' WHERE location IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product SET redirect_type = '301-category' WHERE redirect_type = ''
Gefundene Zeilen: 65
SQL-Befehl:
UPDATE uhqb_product SET available_date = '2020-12-01' WHERE available_date IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product SET cache_default_attribute = '0' WHERE cache_default_attribute IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product_shop SET unity = '' WHERE unity IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product_shop SET redirect_type = '301-product' WHERE redirect_type = ''
Gefundene Zeilen: 59
SQL-Befehl:
UPDATE uhqb_product_shop SET available_date = '2020-12-01' WHERE available_date IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_product_shop SET cache_default_attribute = '0' WHERE cache_default_attribute IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_address SET dni = '' WHERE dni IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_address SET other = '' WHERE other IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_address SET address2 = '' WHERE address2 IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_address SET company = '' WHERE company IS NULL
Gefundene Zeilen: 0
SQL-Befehl:
UPDATE uhqb_address SET vat_number = '' WHERE vat_number IS NULL
Gefundene Zeilen: 0
```


only for redirect type will be changes 🤷‍♀️

I think this will do nothing for me

**Comment by hibatallahAouadni** [comment]
Hello @didi-2018 

Could you please provide more infos cause I didn't manage to reproduce your issue :confounded: 
I see in your screen record that you have a problem with order with customization, is this customization / product deleted?

Please check and feedback.
Thanks!

**Comment by didi-2018** [comment]
@hibatallahAouadni no I have none products with customization.

**Comment by hibatallahAouadni** [comment]
Hello @didi-2018 

Sorry, I meant a product with combinations not customizations :sweat_smile: 
Well, this kind of error usually launched cause you delete a currency or language that you use in these Orders.
And it's odd that you can't enable the debug mode :confounded: 
Could you please try to enable it by setting the constant PS_MODE_DEV to true in config/defines.inc.php.

Please check and feedback.
Thanks!

**Comment by hibatallahAouadni** [comment]
Hello @didi-2018 ,

Since we had no news from you for more than 20 days, I'll close this ticket to avoid cluttering up the backlog.
Please note that you can always create a new one if further information pops up.

Thanks!
